src/env.rs

```diff
@@ -259,7 +265,7 @@ impl Rkv {
     }
 }
 
-#[allow(clippy::cyclomatic_complexity)]
+#[allow(clippy::cognitive_complexity)]
```

Comment:
It looks like this change breaks stable Rust. Can we allow both?

Reply:
I couldn't find a way to get it work both for stable and nightly. Let me see if it makes sense to break those functions down into smaller ones.

Reply:
I ended up swapping the `cyclomatic_complexity` out with a more general `complexity` rule so that it worked both on nightly and stable. We can switch it back once Clippy stable deprecates `cyclomatic_complexity`.